src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -148,28 +156,54 @@ private static void WarnFailedToLoadAssembly(string path, Exception ex)
         {
             Write(
                 ex,
-                "Warning: FakeItEasy failed to load assembly '{0}' while scanning for extension points. Any IArgumentValueFormatters, IDummyFactories, and IFakeOptionsBuilders in that assembly will not be available.",
-                path);
+                $"Warning: FakeItEasy failed to load assembly '{path}' while scanning for extension points. Any IArgumentValueFormatters, IDummyFactories, and IFakeOptionsBuilders in that assembly will not be available.");
         }
 
         private static void WarnFailedToGetTypes(Assembly assembly, Exception ex)
         {
             Write(
                 ex,
-                "Warning: FakeItEasy failed to get types from assembly '{0}' while scanning for extension points. Any IArgumentValueFormatters, IDummyFactories, and IFakeOptionsBuilders in that assembly will not be available.",
-                assembly);
+                $"Warning: FakeItEasy failed to get types from assembly '{assembly}' while scanning for extension points. Any IArgumentValueFormatters, IDummyFactories, and IFakeOptionsBuilders in that assembly will not be available.");
         }
 
-        private static void Write(Exception ex, string messageFormat, params object[] messageArgs)
+        private static void WarnFailedToGetSomeTypes(Assembly assembly, ReflectionTypeLoadException ex)
+        {
+            int notLoadedCount = ex.Types.Count(t => t == null);
+            Write(
+                ex,
+                $"Warning: FakeItEasy failed to get some types from assembly '{assembly}' while scanning for extension points. Some IArgumentValueFormatters, IDummyFactories, and IFakeOptionsBuilders in that assembly might not be available.",
+                writer =>
+                {
+                    using (writer.Indent())
+                    {
+                        writer.Write($"{notLoadedCount} type(s) were not loaded for the following reasons:");
+                        writer.WriteLine();
+                        foreach (var loaderException in ex.LoaderExceptions)
+                        {
+                            writer.Write(" - ");
+                            writer.Write(loaderException.GetType());
+                            writer.Write(": ");
+                            writer.Write(loaderException.Message);
+                            writer.WriteLine();
+                        }
+                    }
+                });
+        }
+
+        private static void Write(Exception ex, string message, Action<IOutputWriter> additionalOutput = null)

```

Comment:
Both feel a little awkward to me.
Would you think it terrible to pass the writer in? e.g.

```c#
var writer = new DefaultOutputWriter(Console.Write);
WriteError(writer, message, ex);
```

in two places, and just above follow it up with

```
using (writer.Indent())
{
    writer.Write($"{notLoadedCount} type(s) were not loaded for the following reasons:");
    writer.WriteLine();
    foreach (var loaderException in ex.LoaderExceptions)
    {
        writer.Write(" - ");
        writer.Write(loaderException.GetType());
        writer.Write(": ");
        writer.Write(loaderException.Message);
        writer.WriteLine();
    }
}
```

?

If that doesn't appeal, I think returning the writer is a little better than the delegate.